src/backend/gl/src/device.rs

```diff
@@ -891,28 +960,81 @@ impl d::Device<B> for Device {
         gl.GenFramebuffers(1, &mut name);
         gl.BindFramebuffer(target, name);
 
-        let att_points = [
-            gl::COLOR_ATTACHMENT0,
-            gl::COLOR_ATTACHMENT1,
-            gl::COLOR_ATTACHMENT2,
-            gl::COLOR_ATTACHMENT3,
-        ];
+        let mut max_color_attachments = 0;
+        gl.GetIntegerv(gl::MAX_COLOR_ATTACHMENTS, &mut max_color_attachments);
+
+        let mut render_attachments = Vec::with_capacity(pass.attachments.len());
+        let mut color_attachment_index = 0;
+        for attachment in &pass.attachments {
+            if color_attachment_index > max_color_attachments as _ {
+                panic!(
+                    "Invalid number of color attachments: {} color_attachment of {}",
+                    color_attachment_index, max_color_attachments
+                );
+            }
+
+            let color_attachment = match color_attachment_index {
+                0 => gl::COLOR_ATTACHMENT0,
```

Comment:
shorter: gl::COLOR_ATTACHMENT0 + index

Reply:
I wasn't sure if all of them were contiguous or not... it makes sense that they would be on every platform.